opentelemetry-api/src/opentelemetry/context/__init__.py

```diff
@@ -104,16 +86,29 @@ def get_current() -> Context:
     return _RUNTIME_CONTEXT.get_current()  # type:ignore
 
 
-def set_current(context: Context) -> Context:
-    """To associate a context with program execution, the Context
-    API provides a function which takes a Context.
+def attach(context: Context) -> object:
+    """Associates a Context with the caller's current execution unit. Returns
+    a token that can be used to restore the previous Context.
+
+    Args:
+        context: The Context to set as current.
+    """
+    get_current()
```

Comment:
I'd suggest to have a `_load_runtime_context()`, a thread safe function that loads the runtime context the **first** time it's called, then call it from `get_current()`, `attach()` and `detach()`. 

It'll clean the code and avoid the call to `_RUNTIME_CONTEXT.get_current()` on `attach()`.

Reply:
To keep things even cleaner, I suggest to implement this `_load_runtime_context` as a decorator and apply it to `get_current`, `attach` and `detach`.

Reply:
added a decorator